rootex/framework/entity_factory.cpp

```diff
@@ -323,28 +318,30 @@ void EntityFactory::deleteEntity(Ref<Entity> entity)
 	entity.reset();
 }
 
-String EntityFactory::saveEntityAsClass(Ref<Entity> entity)
+bool EntityFactory::saveEntityAsClass(Ref<Entity> entity)
 {
-	if (OS::IsDirectory("game/assets/classes/" + entity->getName()))
+	if (OS::IsExists("game/assets/classes/" + entity->getName()))
 	{
 		WARN("Class with name \"" + entity->getName() + "\" already exists. New Class not created.");
-		return nullptr;
+		return false;
 	}
 	else
 	{
 		OS::CreateDirectoryName("game/assets/classes/" + entity->getName());
+		saveEntityAsClassRecursively(entity, "game/assets/classes/" + entity->getName() + "/");
```

Comment:
`saveEntityAsClassRecursively` returns a `String`, I don't think that is needed anymore

Reply:
Used in recursion